### ISSUE TITLE
Feature Request: Installed module introspection to complement live patch module enumeration

### ISSUE THREAD

**Issue by dirkleas** [VCVRack/Rack#926]
Please consider adding a mechanism for rack installation reflection for the installed modules. It would be hugely helpful for installed modules to have JSON metadata in the installation directory as part of the binary distribution that would enumerate the modules, version, features, properties, etc.. This would make it much easier for supplemental tooling to work with .vcv files. For example, it would be extremely helpful to implement/use a scaffolding/utility cli tool to compose multiple patch fragments, identify/correct renamed slugs, perform patch analytics, etc., and ordinal values add to the complexity of such implementations. This would also help user's self-manage modules not under control of the Plugin Manager.

Thanks in advance for your thoughtful consideration.

**Comment by dirkleas** [comment]
can you at least add module width for both patches and installed plugin modules (e.g. gPlugins down at the model level, and gRackWidget down at module). also, please save module with when serializing patches. working code for all [here](https://github.com/dirkleas/Rack), minimally see these two commits: gPlugins-based ["catalog"](https://github.com/dirkleas/Rack/commit/6ff6217a1ac86bb59de099f4bd8e4991aa0b936d), and ModuleWidget [width addition](https://github.com/dirkleas/Rack/commit/812ffdefece9d2f33d67a53030b11b41e806d383) -- the spirit of the fork is to provide a json catalog of all installed plugin modules including width, as well as width  for all modules in saved patches -- see [this](https://github.com/dirkleas/rackcli) for rationale and numerous use cases

**Comment by AndrewBelt** [comment]
Most Rack users don't use a CLI, so the features you mention should be built into Rack itself if implemented at all.

**Comment by dirkleas** [comment]
of course, the CLI is for experimentation and features that are not yet in Rack -- looking forward to retiring CLI features as they appear in standard Rack! w00t!
